packages/module/src/tackle2-ui-legacy/components/TableControls/TableHeaderContentWithBatteries.tsx

```diff
@@ -1,28 +1,48 @@
 import React from 'react';
 import { Th } from '@patternfly/react-table';
+import { TableBatteries } from '../../../types';
 
-export interface TableHeaderContentWithBatteriesProps {
-  numColumnsBeforeData: number;
-  numColumnsAfterData: number;
+export interface TableHeaderContentWithBatteriesProps<
+  TItem,
+  TColumnKey extends string,
+  TSortableColumnKey extends TColumnKey,
+  TFilterCategoryKey extends string = string,
+  TPersistenceKeyPrefix extends string = string
+> extends TableBatteries<TItem, TColumnKey, TSortableColumnKey, TFilterCategoryKey, TPersistenceKeyPrefix> {
   children: React.ReactNode;
 }
 
-export const TableHeaderContentWithBatteries: React.FC<TableHeaderContentWithBatteriesProps> = ({
-  numColumnsBeforeData,
-  numColumnsAfterData,
-  children
-}) => (
-  <>
-    {Array(numColumnsBeforeData)
-      .fill(null)
-      .map((_, i) => (
-        <Th key={i} />
-      ))}
-    {children}
-    {Array(numColumnsAfterData)
-      .fill(null)
-      .map((_, i) => (
-        <Th key={i} />
-      ))}
-  </>
-);
+export const TableHeaderContentWithBatteries = <
+  TItem,
+  TColumnKey extends string,
+  TSortableColumnKey extends TColumnKey,
+  TFilterCategoryKey extends string = string,
+  TPersistenceKeyPrefix extends string = string
+>(
+  props: React.PropsWithChildren<
+    TableHeaderContentWithBatteriesProps<
+      TItem,
+      TColumnKey,
+      TSortableColumnKey,
+      TFilterCategoryKey,
+      TPersistenceKeyPrefix
+    >
+  >
+) => {
+  const { numColumnsBeforeData, numColumnsAfterData, children } = props;
+  return (
+    <>
+      {Array(numColumnsBeforeData)
```

Comment:
I see these were here before, so I'm sure there's a good reason for them, but can you explain the purpose of numColumnsBeforeData and numColumnsAfterData, filling them with null values and rendering headers for each?

Reply:
This is how the Th magically gets Tds in it corresponding to the additional columns that get rendered by `TableRowContentWithBatteries` when features are enabled that add cells for their controls, like the checkbox column for selection or the expansion toggle button column for single-expansion. Based on which features are enabled, the hooks count how many of those columns there are on the left and right sides of the data columns so we can use them to reason about things like colSpan and column indexes.